operator/src/main/resources/scripts/startServer.sh

```diff
@@ -126,11 +147,93 @@ function copySitCfg() {
   fi
 }
 
+# prepare mii server
+
+function prepareMIIServer() {
+
+  trace "Model-in-Image: Creating domain home."
+
+  # primordial domain contain the basic structures, security and other fmwconfig templated info
+  # domainzip only contains the domain configuration (config.xml jdbc/ jms/)
+  # Both are needed for the complete domain reconstruction
+
+  if [ ! -f /weblogic-operator/introspector/primordial_domainzip.secure ] ; then
+    trace "Domain Source Type is FromModel, the primordial model archive is missing, cannot start server"
+    exit 1
+  fi
+
+  if [ ! -f /weblogic-operator/introspector/domainzip.secure ] ; then
+    trace "Domain type is FromModel, the domain configuration archive is missing, cannot start server"
+    exit 1
+  fi
+
+  cd / && base64 -d /weblogic-operator/introspector/primordial_domainzip.secure > /tmp/domain.tar.gz && \
+   tar -xzf /tmp/domain.tar.gz
```

Comment:
add || exitOrLoop  (similar advice applies to all major calls that may fail)

Reply:
Actually, since this is a method, it's more clear to have it return non-zero instead of exit - and then have the caller check the error code and exitOrLoop...

Reply:
@jshum2479 ?

Reply:
Changed to method in PR 1515